### PR TITLE
BUG: Resample BM/BQ adds extra index point #9756

### LINKED ISSUE #9756: Resample BM adds extra index point
```
In [1]: index = pd.DatetimeIndex(start='20150101', end='20150331', freq='B')

In [2]: df = pd.DataFrame(index=index, data=len(index)*[0])

In [3]: df.resample('BM', how='last')
Out[3]:
             0
2015-01-30   0
2015-02-27   0
2015-03-31   0
2015-04-30 NaN
```
This is both in 0.14.1 and 0.15.2

### PR DESCRIPTION
**Pull request by scari**
closes #9756

### CODE REVIEW

File: pandas/tests/test_groupby.py [hunk 1]
```diff
@@ -966,6 +966,15 @@ def demean(arr):
         expected = DataFrame({'b' : range(5)})
         tm.assert_frame_equal(result, expected)
 
+    def test_resample_extra_index_point(self):
+        expected_i = pd.DatetimeIndex(start='20150101', end='20150331', freq='BM')
```
Comment (jreback) [inline pandas/tests/test_groupby.py]: pls add the issue number as a comment here

**Comment by jreback** [comment]
thanks, minor doc point. ping when you have pushed.

**Comment by scari** [comment]
@jreback Thank you for reviewing!

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/c0d4339b18ab29d3d692901e19393a3ea10d7cf3

thanks!

**Pull request closed without merging**